xds: TestRDSHandleResponse is flaky

```
--- FAIL: TestRDSHandleResponse (0.00s)
    rds_test.go:167: Starting fake xDS server at 127.0.0.1:36567...
    rds_test.go:168: Started xDS gRPC client...
    rds_test.go:180: v2c.watchLDS callback, ldsUpdate: {routeName:GoodRouteConfig1}, err: <nil>
    --- FAIL: TestRDSHandleResponse/one-uninteresting-route-config (0.00s)
        rds_test.go:259: v2c.handleRDSResponse() returned err: xds: no RDS watcher found when handling RDS response: resources:<type_url:"type.googleapis.com/envoy.api.v2.RouteConfiguration" value:"\n\020GoodRouteConfig2\0228\022\024uninteresting.domain\032 \n\002\n\000\022\032\n\030UninterestingClusterName\022+\022\017lds.target.good\032\030\n\002\n\000\022\022\n\020GoodClusterName2" > type_url:"type.googleapis.com/envoy.api.v2.RouteConfiguration" , wantErr: false
FAIL
```

--- a/xds/internal/client/rds_test.go
+++ b/xds/internal/client/rds_test.go
@@ -185,6 +185,8 @@ func TestRDSHandleResponse(t *testing.T) {
 	if err := <-cbCh; err != nil {
 		t.Fatalf("v2c.watchLDS returned error in callback: %v", err)
 	}
+	// Read the LDS ack, to clear RequestChan for following tests.
+	<-fakeServer.RequestChan
 
 	tests := []struct {
 		name          string

